packages/wdio-appium-service/src/launcher.js

```diff
@@ -0,0 +1,120 @@
+        let exited = null
+
+        const exitCallback = (exitCode) => {
+            clearTimeout(timer)
+            exited = exitCode
+            callback(new Error(`Appium exited before timeout (exit code: ${exitCode})`), null)
+        }
+
+        const timer = setTimeout(() => {
+            process.removeListener('exit', exitCallback)
+            if (exited === null) {
+                log.debug(`Appium started with ID: ${process.pid}`)
+                callback(null, process)
+            } else {
+                callback(new Error(`Appium exited just after starting (exit code: ${exited})`), null)
+            }
+        }, 5000)
+
+        process.once('exit', exitCallback)
+    }
+
+    _redirectLogStream(logPath) {
+        const logFile = getFilePath(logPath, DEFAULT_LOG_FILENAME)
+
+        // ensure file & directory exists
+        fs.ensureFileSync(logFile)
+
+        log.debug(`Appium logs written to: ${logFile}`)
+        const logStream = fs.createWriteStream(logFile, { flags: 'w' })
+        this.process.stdout.pipe(logStream)
+        this.process.stderr.pipe(logStream)
+    }
+
+    static _getAppiumCommand() {
+        return /^win/.test(process.platform) ? 'appium.cmd' : 'appium'
+    }
+
+    static _cliArgsFromKeyValue(keyValueArgs) {
+        const cliArgs = []
+        for (let key in keyValueArgs) {
+            const value = keyValueArgs[key]
+            // If the value is false or null the argument is discarded
+            if ((typeof value === 'boolean' && !value) || value === null) {
+                continue
+            }
+
+            cliArgs.push(AppiumLauncher._lowerCamelToCliOptionName(key))
+
+            // Only non-boolean and non-null values are added as option values
+            if (typeof value !== 'boolean' && value !== null) {
+                cliArgs.push(AppiumLauncher._sanitizeCliOptionValue(value))
+            }
+        }
+        return cliArgs
+    }
+
+    static _lowerCamelToCliOptionName(camelCasedKey) {
+        let cliOptionName = '--'
+        const bigACharCode = 'A'.charCodeAt(0)
+        const bigZCharCode = 'Z'.charCodeAt(0)
+        for (let charIndex = 0; charIndex < camelCasedKey.length; ++charIndex) {
+            const char = camelCasedKey.charAt(charIndex)
+            const charCode = camelCasedKey.charCodeAt(charIndex)
+            // If the character between A and Z replace it with a dash and a small char
+            if (charCode >= bigACharCode && charCode <= bigZCharCode) {
+                cliOptionName += `-${char.toLowerCase()}`
+            } else {
+                cliOptionName += char
+            }
+        }
+        return cliOptionName
+    }
+
+    static _sanitizeCliOptionValue(value) {
+        const valueString = String(value)
+        // Encapsulate the value string in single quotes if it contains a white space
+        return /\s/.test(valueString) ? `'${valueString}'` : valueString
+    }
+}
```

Comment:
This is more of a question than anything else, but isn't it a convention to have static methods defined before the constructor? (not like it matters much other than just readability)

Reply:
The methods didn't need to be static so I removed it. I think they should be in the bottom as they are helpers